pkg/testutils/testcluster/testcluster.go

```diff
@@ -317,6 +317,9 @@ func NewTestCluster(
 	if clusterArgs.StartSingleNode && nodes > 1 {
 		t.Fatal("StartSingleNode implies 1 node only, but asked to create", nodes)
 	}
+	if clusterArgs.ServerArgs.ClusterName == "" {
+		clusterArgs.ServerArgs.ClusterName = fmt.Sprintf("TestCluster-%d", timeutil.Now().UnixNano())
```

Comment:
On macOS you don't really get nanosecond resolution so it is a bit easier to get a timestamp conflict.

We could put a random slug as well.

Reply:
Can just use rand.Uint64() (from rand/v2).

Reply:
Yeah, I like `rand` better than timestamps, because the issue fixed here happens when two clusters are close in time.